alghaibb/al-haya
# Add toggle password visibility and loading spinner to Signup form

### Description:

This pull request introduces several enhancements to the signup process, improving both user experience and interface responsiveness. The key features added include:

1.  **Password Visibility Toggle:** Users can now toggle the visibility of their password and confirmation password, enhancing usability, especially on mobile devices. This feature is implemented using eye/eye-off icons from the Ant Design Icons library.

2.  **Loading Spinner:** A spinner animation is displayed when the signup button is clicked, providing immediate feedback to the user that their signup request is being processed. This helps prevent multiple submissions and improves user interaction.

3.  **Delayed Toast Notifications:** Toast messages for successful registration and error handling now include a delay, ensuring that users have adequate time to read and understand the message.

4.  **CSS Adjustments:** Additional CSS styling has been added to position the password visibility toggle icon inside the input field, aligning with modern design practices.

### Technical Details:

-   Utilized `useState` hooks for managing the password visibility state.
-   Integrated the `EyeInvisibleOutlined` and `EyeOutlined` icons for the toggle feature.
-   Implemented a new CSS class `.form-item` to handle the positioning of the toggle button within the form.
-   Added `loadingSpinner` component styled with keyframe animations for a smooth spinning effect.

### Tests:

-   Manual testing was conducted to ensure the toggle feature works as expected across different screen sizes.
-   Verified that the loading spinner displays correctly during the signup process and disappears once complete.
-   Toast notifications were tested for both successful and failed registration attempts.

### How to Test:

1.  Navigate to the signup page.
2.  Enter details and observe the password toggle functionality.
3.  Click the signup button and notice the loading spinner.
4.  Check toast messages for both successful and unsuccessful signup attempts.
